c_formatter_42/__main__.py

```diff
@@ -31,34 +76,30 @@ def main() -> int:
         help="Ask confirmation before overwriting any file",
     )
     arg_parser.add_argument(
-        "filepaths",
-        metavar="FILE",
+        "-i",
+        "--ignore",
+        nargs='+',
+        default=[],
+        help="Ignore specified folders (e.g. .git/ .vscode/)",
+    )
+    arg_parser.add_argument(
+        "paths",
+        metavar="PATH",
         nargs="*",
-        help="File to format inplace, if no file is provided read STDIN",
+        help="Files or directories to format. If no path is provided, read STDIN",
     )
     args = arg_parser.parse_args()
-
-    if len(args.filepaths) == 0:
+    
+    if len(args.paths) == 0:
         content = sys.stdin.read()
         print(run_all(content), end="")
         return 0
-
-    for filepath in args.filepaths:
-        try:
-            with open(filepath, "r") as file:
-                content = file.read()
-            if args.confirm:
-                result = input(f"Are you sure you want to overwrite {filepath}?[y/N]")
-                if result != "y":
-                    continue
-            print(f"Writing to {filepath}")
-            with open(filepath, "w") as file:
-                file.write(run_all(content))
-        except OSError as e:
-            print(f"Error: {e.filename}: {e.strerror}", file=sys.stderr)
-            return 1
-    return 0
-
+    
+    success = True
+    for path in args.paths:
+        success &= process_path(path, args.confirm, args.ignore)
```

Comment:
```suggestion
    success = all(process_path(p, args.confirm, args.ignore) for p in args.path)
```

Same for the `success` code in the function `process_path`